src/modm/driver/dac/ma12070p.hpp

```diff
@@ -0,0 +1,354 @@
+		LimiterCh1R		= Bit3,
+		ClippingCh0L	= Bit4,
+		ClippingCh0R	= Bit5,
+		ClippingCh1L	= Bit6,
+		ClippingCh1R	= Bit7,
+	};
+	MODM_FLAGS8(VlpMonitor);
+
+#if MODM_HAS_IOSTREAM
+	friend IOStream&
+	operator << (IOStream& os, const VlpMonitor_t& c)
+	{
+		os << "VlpMonitor(";
+		if(c & VlpMonitor::LimiterCh0L)
+			os << "LimiterCh0L ";
+		if(c & VlpMonitor::LimiterCh0R)
+			os << "LimiterCh0R ";
+		if(c & VlpMonitor::LimiterCh1L)
+			os << "LimiterCh1L ";
+		if(c & VlpMonitor::LimiterCh1R)
+			os << "LimiterCh1R ";
+		if(c & VlpMonitor::ClippingCh0L)
+			os << "ClippingCh0L ";
+		if(c & VlpMonitor::ClippingCh0R)
+			os << "ClippingCh0R ";
+		if(c & VlpMonitor::ClippingCh1L)
+			os << "ClippingCh1L ";
+		if(c & VlpMonitor::ClippingCh1R)
+			os << "ClippingCh1R ";
+		os << ")";
+		return os;
+	}
+#endif
+
+	enum class
+	ErrorRegister : uint8_t
+	{
+		FlyingCapOverVolt		= Bit0,
+		OverCurrent				= Bit1,
+		Pll						= Bit2,
+		PvddUnderVolt			= Bit3,
+		OverTempWarning			= Bit4,
+		OverTempError			= Bit5,
+		PinToPinLowImpedance	= Bit6,
+		DcProtection			= Bit7,
+	};
+	MODM_FLAGS8(ErrorRegister);
+
+#if MODM_HAS_IOSTREAM
+	friend IOStream&
+	operator << (IOStream& os, const ErrorRegister_t& c)
+	{
+		os << "ErrorRegister(";
+		if(c & ErrorRegister::FlyingCapOverVolt)
+			os << "FlyingCapOverVolt ";
+		if(c & ErrorRegister::OverCurrent)
+			os << "OverCurrent ";
+		if(c & ErrorRegister::Pll)
+			os << "Pll ";
+		if(c & ErrorRegister::PvddUnderVolt)
+			os << "PvddUnderVolt ";
+		if(c & ErrorRegister::OverTempWarning)
+			os << "OverTempWarning ";
+		if(c & ErrorRegister::OverTempError)
+			os << "OverTempError ";
+		if(c & ErrorRegister::PinToPinLowImpedance)
+			os << "PinToPinLowImpedance ";
+		if(c & ErrorRegister::DcProtection)
+			os << "DcProtection ";
+		os << ")";
+		return os;
+	}
+#endif
+
+
+	// quarter_decibel type
+	enum class
+	quarter_decibel_t : int16_t
+	{};
+
+	template<typename T>
+	static constexpr quarter_decibel_t
+	quarter_decibel(T value)
+	{
+		return quarter_decibel_t(int16_t(value * 4ul));
+	}
+	
+	static constexpr float
+	quarterDecibelToFloat(quarter_decibel_t value)
+	{
+		return int16_t(value) / 4.f;
+	}
+
+	static constexpr quarter_decibel_t MaxVolume = static_cast<quarter_decibel_t>(24*4);
+	static constexpr quarter_decibel_t MinVolume = static_cast<quarter_decibel_t>(-144*4);
+	// static constexpr quarter_decibel_t MaxVolume = quarter_decibel(24);
+	// static constexpr quarter_decibel_t MinVolume = quarter_decibel(-144);
+};  // struct ma12070p
+
+constexpr ma12070p::quarter_decibel_t
+operator-(ma12070p::quarter_decibel_t v)
+{
+	return static_cast<ma12070p::quarter_decibel_t>(-static_cast<int16_t>(v));
+}
+
+namespace literals
+{
+	constexpr ma12070p::quarter_decibel_t
+	operator""_q_db(long double value)
+	{
+		return ma12070p::quarter_decibel(value);
+	}
+
+	constexpr ma12070p::quarter_decibel_t
+	operator""_q_db(unsigned long long int value)
+	{
+		return ma12070p::quarter_decibel(value);
+	}
```

Comment:
I implemented the quarter decibel data type mainly to make it convenient for the user to use decibels in the application code without having to look up the meaning of otherwise unitless parameters for `setMasterVolume()`, etc...